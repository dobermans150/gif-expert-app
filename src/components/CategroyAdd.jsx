import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategroyAdd = ({ setcategories }) => {
  const [inputValue, setinputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length) {
      setcategories((categories) => [inputValue, ...categories]);
      setinputValue('');
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

CategroyAdd.propTypes = {
  setcategories: PropTypes.func.isRequired,
};

export default CategroyAdd;
