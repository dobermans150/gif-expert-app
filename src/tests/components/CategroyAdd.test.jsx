import React from 'react';
import { shallow } from 'enzyme';
import CategroyAdd from '../../components/CategroyAdd';

describe('Pruebas en el componentes CategoryAdd', () => {
  const setCategories = () => {};
  const wrapper = shallow(<CategroyAdd setcategories={setCategories} />);

  test('Debe dem ostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });
  });
});
