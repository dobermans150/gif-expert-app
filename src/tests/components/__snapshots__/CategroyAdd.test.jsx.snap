// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes CategoryAdd Debe dem ostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
